src/plugins/discover/public/application/components/discover_grid/discover_grid.tsx

```diff
@@ -288,11 +320,11 @@ export const DiscoverGrid = ({
 
   if (!rowCount) {
     return (
-      <div className="euiDataGrid__noResults">
+      <div className="euiDataGrid__loading">
         <EuiText size="xs" color="subdued">
-          <EuiIcon type="discoverApp" size="m" color="subdued" />
+          <EuiLoadingSpinner />
           <EuiSpacer size="s" />
-          <FormattedMessage id="discover.noResultsFound" defaultMessage="No results found" />
+          <FormattedMessage id="discover.loadingResults" defaultMessage="Loading results" />
         </EuiText>
```

Comment:
Thanks for adapting this. On change request here. the current implementation replaces no results with loading. But there's a **loading state** AND a **no results state** , the loading state should be displayed when `!rowCount` and  `isLoading` is true, else the no results state should be displayed.